ddtrace/tracer/sampler.go

```diff
@@ -24,6 +24,36 @@ import (
 	"golang.org/x/time/rate"
 )
 
+// samplerName specifies the identification of a sampler which was
+// responsible for a certain sampling decision.
+type samplerName int8
+
+const (
+	// samplerNone specifies that the sampling decision was already made
+	// and it's noop except propagating the decision.
```

Comment:
And what "is a noop"? Maybe it's best to remove the last part, or call this `samplerLocked`. I'm not sure I completely understand the purpose of it.

Reply:
I'm using it when the sampling priority is from an upstream service. So, we can know that we don't need to build `_dd.p.upstream_services` tag from it. I can call it `samplerUpstream` if it makes more sense.

Reply:
`samplerUpstream` is clearer if this always indicates it was sampled upstream.